app/src/main/java/org/simple/clinic/patient/PatientRepository.kt

```diff
@@ -493,9 +487,26 @@ class PatientRepository @Inject constructor(
         }
 
     return businessIdStream
-        .flatMap { businessId ->
-          Completable.fromAction { database.businessIdDao().save(listOf(businessId)) }
-              .toSingleDefault(businessId)
+        .flatMap { businessId -> saveBusinessId(businessId).toSingleDefault(businessId) }
+        .flatMap { businessId -> touchPatientRecord(patientUuid).toSingleDefault(businessId) }
+  }
+
+  private fun touchPatientRecord(patientUuid: UUID): Completable {
```

Comment:
Can you remove this method entirely and call `setSyncStatus` from `addIdentifierToPatient`. That will internally calls `database.patientDao().updateSyncStatus`?

Reply:
Done.